[Doc] Add substr function des

francisoliverlee opened the pull request:
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [] Bugfix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation Update (if none of the other choices apply)
- [] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [] I have create an issue on (Fix #ISSUE), and have described the bug/feature there in detail
- [] Compiling and unit tests pass locally with my changes
- [] I have added tests that prove my fix is effective or that my feature works
- [] If this change need a document change, I have updated the document
- [] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

Commit:
Merge pull request #1 from apache/master

Pull from Apache

Commit:
add substr function doc

Commit:
add substr function doc

morningman commented on docs/en/sql-reference/sql-functions/string-functions/substr.md:
````diff
@@ -0,0 +1,41 @@
+| o d                         |
++-----------------------------+
+```
+##keyword
````
```suggestion
## keyword
```

morningman commented on docs/zh-CN/sql-reference/sql-functions/string-functions/substr.md:
````diff
@@ -0,0 +1,40 @@
+| o d                         |
++-----------------------------+
+```
+##keyword
````
```suggestion
## keyword
```

xy720 committed:
[Spark Load] [Bug] Load job's state will stay in ETL state all the time after FE restart  (#4528)

morningman committed:
[Bug] Fix bug that table ids is not set right for hadoop load job (#4535)

We store all table ids involved in the Load Job in TransactionState.
However, for Hadoop Load job, table ids are set incorrectly.

This caused the WAITING_TXN phase to not correctly wait for the completion
of the previous load transaction when doing the alter table,
which caused some data version loss problems.

ZhangYu0123 committed:
[Log] Add log for trace broker (#4505)

Add tracing broker log. When fe get filestatus for distributing load task to broker,
the broker maybe get empty files and not give correct error code.
Add this log to easy track which broker process filestatus operation and we can get the error log.

ZhangYu0123 committed:
[BUG] Tablet is not readable and delete handler report -1903 error, when condition value contains \n (#4531)

xy720 committed:
[Spark load][Bug] fix that cancelling a spark load in the `PENDING` phase will not succeed (#4536)

xy720 committed:
[Spark Load][Bug] Keep the column splitting in spark load consistent with broker load / mini load (#4532)

zhaojintaozhao committed:
[Variable] Add a new  global variable  "performance_schema" to support Mysql JDBC 8.0.16 or later.#4537#  (#4542)

ZhangYu0123 committed:
[BUG] Fix stale path delete checking logic when current main path is missing. (#4549)

Fix stale path delete checking logic.
When current main path is version missing, then delete checking logic is always core dumped. So we fix the checking logic to tolerate current main version missing.

EmmyMiao87 committed:
[Bug] Fix null pointer when replay RollupJob (#4572)

When fe replay rollup job(v1) with deleted table, it will throw null pointer exception and exit.
This commit ignore this error and print a warning log to avoid fe exit.

Fixed #4571

Change-Id: I302b554a94d42aee645db6b224cd989e00cd3ca6

xinghuayu007 committed:
[Bug] catch plugin init error for Reinstall plugin (#4561)

vagetablechicken committed:
[Bug] fix get_parsed_paths() subscript out of range (#4585)

wangbo committed:
[SparkDpp] Support complete types (#4524)

For[Spark Load]
1 support decimal andl largeint
2 add validate logic for char/varchar/decimal
3 check data load from hive with strict mode
4 support decimal/date/datetime aggregator

caiconghui committed:
[Bug]Fix bug that tablet info with wrong capacity may cause fe oom (#4567)

* [Bug]Fix bug that tablet info with wrong capacity may cause fe oom

* remove unused code

* fix

* remove empty line

Co-authored-by: caiconghui [蔡聪辉] <caiconghui@xiaomi.com>

wutiangan committed:
[SQL][Planner] Fix the the parallesim of fragment which has 3 or more childern #4569 (#4570)

fix the the parallelism of fragment which has 3 or more childern

EmmyMiao87 committed:
[BUG] ReAgg when adding agg mv on dup base table (#4587)

When the keystype of mv and base table is difference, Doris should execute
sorting schema change instead of linked schema change.
If doesn't, the data size of mv actually is same as base table.
This will cause mv to have no pre-aggregation effect at all.
The query will not choose mv.

This commit fixed this problem. Fixed #4586

morningman-cmy committed:
[compaction][config] Change default config policy to size_based

Change-Id: Id6a1023cb9b4b23bb0714572aeca5d8a0c9ce865

morningman committed:
[Compile] Add pluginRepository for java-cup-plugins (#4636)

the cup-maven-plugin of net.sourceforge.czt.dev is missing in maven central repo.
It has been moved to https://repository.cloudera.com/content/groups/public/

Co-authored-by: morningman <chenmingyu@baidu.com>

EmmyMiao87 committed:
[License] Add license header (#4664)

yangzhg committed:
[Compile] Fix md5 of thirdparty(#4663)

Fix compile error when compile Doris from source and not using docker

Commit:
Merge branch 'master' of https://github.com/apache/incubator-doris into add-substr-functionn-des

Commit:
update
